File: api/src/main/java/com/api/app/emum/PRD001.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 상품상태코드: goodsStatusCode
 * */
@Getter
@AllArgsConstructor
public enum PRD001 implements CommonCode {
    ON_SALE("001", "판매중", 1, "", ""),
    SALE_STOPPED("002", "판매중단", 2, "", ""),
    SALE_SUSPENDED("003", "판매중지", 3, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static PRD001 findByCode(String code) {
        return CommonCodeUtil.findByCode(PRD001.class, code);
    }
}

File: api/src/main/kotlin/com/api/app/emum/PRD001.kt
package com.api.app.emum

/** 상품상태코드: goodsStatusCode */
enum class PRD001(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    ON_SALE("001", "판매중", 1, "", ""),
    DISCONTINUED("002", "판매중단", 2, "", ""),
    SOLD_OUT("003", "품절", 3, "", ""),
    ;

    companion object {
        fun findByCode(code: String): PRD001? = CommonCodeUtil.findByCode(PRD001::class.java, code)
    }
}
